checkov/kubernetes/image_referencer/provider/k8s.py

```diff
@@ -19,77 +19,55 @@ def __init__(self, graph_connector: DiGraph) -> None:
-
-        containers = spec.get("initContainers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-    return image_names
+    return _extract_images_from_spec(spec)
 
 
 def extract_images_from_pod(resource: dict[str, Any]) -> list[str]:
-    image_names: list[str] = []
-
     spec = resource.get("spec")
-    if isinstance(spec, dict):
-        containers = spec.get("containers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-        containers = spec.get("initContainers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-    return image_names
+    return _extract_images_from_spec(spec)
 
 
 def extract_images_from_pod_template(resource: dict[str, Any]) -> list[str]:
     # the 'PodTemplate' object is usually not defined by the user, but rather used by Kubernetes internally
-    image_names: list[str] = []
-
     spec = find_in_dict(input_dict=resource, key_path="template/spec")
-    if isinstance(spec, dict):
-        containers = spec.get("containers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-        containers = spec.get("initContainers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-    return image_names
+    return _extract_images_from_spec(spec)
 
 
 def extract_images_from_template(resource: dict[str, Any]) -> list[str]:
-    image_names: list[str] = []
-
     spec = find_in_dict(input_dict=resource, key_path="spec/template/spec")
-    if isinstance(spec, dict):
-        containers = spec.get("containers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-        containers = spec.get("initContainers")
-        image_names.extend(extract_images_from_containers(containers=containers))
-
-    return image_names
+    return _extract_images_from_spec(spec)
 
 
-def extract_images_from_containers(containers: Any) -> list[str]:
+def extract_images_from_containers(containers: Any) -> set[str]:
     """Helper function to extract image names from containers block"""
 
-    image_names: list[str] = []
+    image_names: set[str] = set()
 
     if isinstance(containers, list):
         for container in containers:
             if isinstance(container, dict):
                 image = container.get("image")
                 if image and isinstance(image, str):
-                    image_names.append(image)
+                    image_names.add(image)
 
     return image_names
 
 
+def _extract_images_from_spec(spec: dict[str, Any] | None) -> list[str]:
+    image_names: set[str] = set()
+
+    if isinstance(spec, dict):
+        containers = spec.get("containers")
+        image_names.update(extract_images_from_containers(containers=containers))
+
+        containers = spec.get("initContainers")
+        image_names.update(extract_images_from_containers(containers=containers))
```

Comment:
maybe update only if containers is not None ?

Reply:
it just adds some more code and makes it look more ugly :/